pkg/cloud/azureprovider.go

```diff
@@ -1208,35 +1210,46 @@ func (az *Azure) UpdateConfig(r io.Reader, updateType string) (*CustomPricing, e
 			c.AzureContainerPath = asc.ContainerPath
 			c.AzureCloud = asc.AzureCloud
 		} else {
-			defer az.DownloadPricingData()
+			// This will block if not in a goroutine. It calls GetConfig(), which
+			// in turn calls GetCustomPricingData, which acquires the same lock
+			// that is acquired by az.Config.Update, which is the function to
+			// which this function gets passed, and subsequently called. Booo.
+			defer func() {
+				go az.DownloadPricingData()
+			}()
```

Comment:
This is the fix. Everything else is error wrapping, gofmt, etc.

Reply:
Not sure how to best protect ourselves against reentrant locking, or why golang doesn't support this. Provider APIs need a major face lift.